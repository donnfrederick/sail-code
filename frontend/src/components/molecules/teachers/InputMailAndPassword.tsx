import InputMailAddress from 'components/atoms/teachers/InputMailAddress'
import InputPassword from 'components/atoms/teachers/InputPassword'
import { Info } from 'models/teachers'
import * as React from 'react'
import styled from 'styled-components'

interface Props {
  error?: any
  info: Info
  marginBottom?: number
  register(): void
}

export default (props: Props) => {
  const { error, info, marginBottom = 80, register } = props
  return (
    <Container marginBottom={marginBottom}>
      <InputMailAddress error={error} info={info} register={register} />
      <InputPassword error={error} info={info} register={register} />
    </Container>
  )
}

const Container = styled<Props, any>('div')`
  width: 640px;
  margin: 0 auto ${props => props.marginBottom}px;
`
