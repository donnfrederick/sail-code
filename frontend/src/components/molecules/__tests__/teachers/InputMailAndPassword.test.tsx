import InputMailAndPassword from 'components/molecules/teachers/InputMailAndPassword'
import * as React from 'react'
import * as renderer from 'react-test-renderer'

test('render <InputMailAndPassword />', () => {
  const info = {
    email: '',
    password: '',
    name: '',
    gender: '',
    picture: '',
    hobbies: [],
    purposes: [],
    desiredCondition: '',
    introduce: '',
    categories: [],
    tags: [],
    selected_tags: [],
    added_tags: []
  }
  const tree = renderer
    .create(<InputMailAndPassword info={info} register={() => null} />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
