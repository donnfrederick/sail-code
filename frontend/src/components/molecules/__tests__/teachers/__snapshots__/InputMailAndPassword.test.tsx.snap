// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render <InputMailAndPassword /> 1`] = `
<div
  className="sc-bZQynM bCUdom"
>
  <div
    className="sc-bdVaJa SRmkz"
  >
    <div
      className="sc-bwzfXH CfcXv"
      data-error={false}
    >
      メールアドレス
    </div>
    <input
      className="sc-htpNat NsTik"
      data-error={false}
      defaultValue=""
      maxLength={254}
      name="mail"
      onInput={[Function]}
      placeholder="sail@example.com"
    />
  </div>
  <div
    className="sc-bxivhb hEzhDS"
  >
    <div
      className="sc-ifAKCX hYHdCw"
      data-error={false}
    >
      パスワード
    </div>
    <input
      className="sc-EHOje SoFRe"
      data-error={false}
      defaultValue=""
      maxLength={72}
      onInput={[Function]}
      pattern="^[0-9A-Za-z]+$"
      placeholder="半角英数8字以上"
      type="password"
    />
  </div>
</div>
`;
